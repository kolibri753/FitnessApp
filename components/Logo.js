import React from "react";

const Logo = () => {
  const logoTextStyles = [
		styles.logoText,
	];

	return (
		<View style={styles.logoContainer}>
			<Text style={logoTextStyles}>Gym</Text>
			<Logo width={60} height={40} fill="white" />
			<Text style={[logoTextStyles, { color: colors.white }]}>Rat</Text>
		</View>
	);
};

const styles = StyleSheet.create({
	logoContainer: {
		flexDirection: "row",
		alignItems: "center",
	},
	logoText: {
		fontSize: 30,
		fontWeight: "bold",
		color: colors.yellow,
	},
});

export default Logo;
