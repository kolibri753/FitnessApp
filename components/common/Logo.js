import React from "react";
import { StyleSheet, View, Text } from "react-native";
import { colors } from "../../styles/colors";
import LogoImg from "../../assets/logo.svg";

const Logo = ({ style }) => {
	return (
		<View style={[styles.logoContainer, style]}>
			<Text style={styles.logoText}>Gym</Text>
			<LogoImg width={60} height={40} fill="white" testID="logoSVG" />
			<Text style={[styles.logoText, { color: colors.white }]}>Rat</Text>
		</View>
	);
};

const styles = StyleSheet.create({
	logoContainer: {
		flexDirection: "row",
		alignItems: "center",
	},
	logoText: {
		fontSize: 30,
		fontWeight: "bold",
		color: colors.yellow,
	},
});

export default Logo;
