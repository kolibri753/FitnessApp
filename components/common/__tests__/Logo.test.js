import React from "react";
import { render } from "@testing-library/react-native";
import renderer from "react-test-renderer";
import Logo from "../Logo";

// Mock the SVG component
jest.mock("../../../assets/logo.svg", () => "SvgMock");

describe("Logo", () => {
	it("renders correctly", () => {
		const { getByText, getByTestId } = render(<Logo />);

		// Check if "Gym" text is rendered
		expect(getByText("Gym")).toBeTruthy();

		// Check if SVG logo is rendered
		const logoSvg = getByTestId("logoSVG");
		expect(logoSvg).toBeTruthy();

		// Check if "Rat" text is rendered
		expect(getByText("Rat")).toBeTruthy();
	});

	it("renders correctly", () => {
		const tree = renderer.create(<Logo />).toJSON();
		expect(tree).toMatchSnapshot();
	});
});
