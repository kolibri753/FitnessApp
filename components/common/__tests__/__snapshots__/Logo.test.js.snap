// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo renders correctly 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "flexDirection": "row",
      },
      undefined,
    ]
  }
>
  <Text
    style={
      {
        "color": "#FFD500",
        "fontSize": 30,
        "fontWeight": "bold",
      }
    }
  >
    Gym
  </Text>
  <SvgMock
    fill="white"
    height={40}
    testID="logoSVG"
    width={60}
  />
  <Text
    style={
      [
        {
          "color": "#FFD500",
          "fontSize": 30,
          "fontWeight": "bold",
        },
        {
          "color": "#FFFFFF",
        },
      ]
    }
  >
    Rat
  </Text>
</View>
`;
